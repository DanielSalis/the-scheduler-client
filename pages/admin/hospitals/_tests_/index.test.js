import { shallowMount, createLocalVue } from '@vue/test-utils'
import Vuetify from 'vuetify'
import AdminHospitalsPage from '../index.vue'
import hospitalsList from '~/components/hospitals/hospitals-list.vue'
import hospitalsAdd from '~/components/hospitals/hospitals-add.vue'

// Mock the layout
const MockLayout = {
  template: '<div><slot /></div>',
}

describe('AdminHospitalsPage.vue', () => {
  let localVue
  let vuetify

  beforeEach(() => {
    localVue = createLocalVue()
    vuetify = new Vuetify()
    localVue.use(Vuetify)
  })

  it('renders the component and matches snapshot', () => {
    const wrapper = shallowMount(AdminHospitalsPage, {
      localVue,
      vuetify,
      mocks: {
        $axios: {
          get: jest.fn(() => Promise.resolve({ data: [] })),
        },
      },
      stubs: {
        layout: MockLayout,
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('computes headers correctly', () => {
    const wrapper = shallowMount(AdminHospitalsPage, {
      localVue,
      vuetify,
      mocks: {
        $axios: {
          get: jest.fn(() => Promise.resolve({ data: [] })),
        },
      },
      stubs: {
        layout: MockLayout,
      },
    })

    const expectedHeaders = [
      { text: 'id', value: 'id' },
      { text: 'Nome', value: 'name' },
      { text: 'Endere√ßo', value: 'address' },
      { text: 'Rede hospitalar', value: 'company.name' },
      { text: 'ACTIONS', value: 'actions', sortable: false },
    ]

    expect(wrapper.vm.headers).toEqual(expectedHeaders)
  })
})
