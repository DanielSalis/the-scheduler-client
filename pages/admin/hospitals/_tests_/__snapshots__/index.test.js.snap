// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminHospitalsPage.vue renders the component and matches snapshot 1`] = `
"<gcontainer-stub class=\\"admin-companies-page-container\\">
  <v-card-stub loaderheight=\\"4\\" tag=\\"div\\">
    <v-toolbar-stub color=\\"white\\" tag=\\"header\\" extensionheight=\\"48\\" flat=\\"true\\" src=\\"\\">
      <v-toolbar-title-stub tag=\\"div\\">Hospitais</v-toolbar-title-stub>
    </v-toolbar-stub>
    <v-tabs-stub activeclass=\\"\\" nexticon=\\"$next\\" previcon=\\"$prev\\" slidersize=\\"2\\">
      <v-tab-stub ripple=\\"true\\">
        <v-icon-stub left=\\"\\">
          mdi-table
        </v-icon-stub>
        Listar
      </v-tab-stub>
      <v-tab-stub ripple=\\"true\\">
        <v-icon-stub left=\\"\\">
          mdi-plus
        </v-icon-stub>
        Adicionar
      </v-tab-stub>
      <v-tab-item-stub>
        <!---->
      </v-tab-item-stub>
      <v-tab-item-stub>
        <hospitalsadd-stub></hospitalsadd-stub>
      </v-tab-item-stub>
    </v-tabs-stub>
  </v-card-stub>
</gcontainer-stub>"
`;
